lib/recorder.js

```diff
@@ -42,19 +43,20 @@ class ConsoleRecorder extends NoOpRecorder {
 }
 
 class HttpRecorder {
-    constructor(url, username, password, debug) {
+    constructor(url, username, password, debug = false, options = {}) {
         function auth(name, pass) {
             const namePass = `${name}:${pass}`;
             return `Basic ${new Buffer(namePass).toString('base64')}`;
         }
 
-        this[HTTP_OPTIONS] = {
-            endpoint: `${url}/hawkular/apm/traces/fragments`,
-            headers: {
-                Authorization: auth(username, password),
-                'Content-Type': 'application/json',
-            },
-        };
+        this[ENDPOINT] = `${url}/hawkular/apm/traces/fragments`;
+        this[HTTP_OPTIONS] = Object.assign({
```

Comment:
`Object.assign` is older, you should now use ` spread operator`, advantage is that it will create new object of options and is somewhat more readable. However it needs to be polyfilled with babel, [es6 proposal](https://github.com/sebmarkbage/ecmascript-rest-spread).
```javascript
this[HTTP_OPTIONS] = {method: 'POST', ...options}
```